### ISSUE TITLE
Achievements markup support

### ISSUE THREAD

**Issue by hladkyjakub** [wesnoth/wesnoth#9410]
### Game and System Information

- **Version:**1.18.0

### Description of the bug

When you complete an achievement, you see a dialog with propper markup support.
![Screenshot_20241001_141859](https://github.com/user-attachments/assets/9f38b056-3677-41f3-a82b-85bc1dac61ee)
But The same text appears wrongly in the achievements menu.
![Screenshot_20241001_141928](https://github.com/user-attachments/assets/48eaaad9-0a87-4c86-97e2-78bb48d5856e)

### Steps to reproduce the behavior

1. Find your favorite achievement, add "<span color='#FF00FF'>This should be pink </span>" to the name or description of the achievement.
2. View the achievement in-game.

### Expected behavior

Propper rainbow text in achievement menu.

**Comment by hladkyjakub** [comment]
> This was an intentional decision to disallow markup in the achievement title, to ensure that there is at least one consistent way that allows a player to distinguish complete from incomplete achievements.

It's distinguished in icon.
But I put the markup only to the completed title, which according to wiki should allow markup.

**Comment by CelticMinstrel** [comment]
> It's distinguished in icon.

This isn't guaranteed as far as I know.

**Comment by hladkyjakub** [comment]
> This isn't guaranteed as far as I know.

If you use grayscaled icon, then no.

But the point was that this is name_completed, why this couln't be corolful?

**Comment by hladkyjakub** [comment]
> > It's distinguished in icon.
> 
> This isn't guaranteed as far as I know.

Is there reason, why completed title couln't be colorfull?

**Comment by CelticMinstrel** [comment]
Yes. We want to prevent an author from creating an achievement that looks completed when it's not or that looks incomplete when it's complete. Disallowing markup in the title means that, no matter what crazy thing the author might do, you can always tell from the title whether the achievement is completed.

**Comment by hladkyjakub** [comment]
> Yes. We want to prevent an author from creating an achievement that looks completed when it's not or that looks incomplete when it's complete. Disallowing markup in the title means that, no matter what crazy thing the author might do, you can always tell from the title whether the achievement is completed.

But ... whats that for?
If someone is dumb enought to make dumb achievements for his add-on, it's just his add-on what is bad, it doesn't affect rest of the game.
That's like you theoretically *can* make an add-on that won't load/will crush the game/won't do literaly anything...
So if someones add-on would have such achoevements, (no idea why would someome do that, it's still much better than crashing game after you open the game for example).
And I'd really like to have achievements about hoarding gold golden once completed, achievement for beta tester with neon can to have rainbow-neon title, achievement for planting trees green. And so on.
You see what I mean?
